### PR TITLE
Rest Advanced module refactoring

### PR DESCRIPTION
**Pull request by Evelina02**

### CODE REVIEW

File: stage #4/module #2. REST API Advanced/rest_api_advanced.md [hunk 1]
```diff
@@ -18,105 +18,124 @@
...
+     * by Id
+     * sort by Created, Modified Asc/Desc. Default: Created Desc
+   * CRD operations for Tag, and Author
+   * Get Tags and Authors:
+     * get all
+     * by Id
+     * by part Name
+     * by News Id (URL examples:\
+           /news/{newsId}/tags - *should return tags collection*\
+           /news/{newsId}/authors - *should return 1 author*)
```
Comment (AnastasiyaTrafimava) [inline stage #4/module #2. REST API Advanced/rest_api_advanced.md:54]: if return 1 author, that need /news/{newsId}/author

Comment (Evelina02) [reply stage #4/module #2. REST API Advanced/rest_api_advanced.md]: updated

**Review by AnastasiyaTrafimava — APPROVED** [review]

File: stage #4/module #2. REST API Advanced/rest_api_advanced.md [hunk 2]
```diff
@@ -18,105 +18,124 @@
...
+##### Business requirements
+
+1. Develop web service for News Management system with the following entities:
+   ![Schema](./media/schema.jpg)
+   * <img src="./media/snowflake.jpg" alt="Snowflake" width="18"/> - unique value
+   * All Name, Title and Content fields are required
+   * Created, Modified – have ISO 8601 format [wiki: ISO_8601](https://en.wikipedia.org/wiki/ISO_8601). Example: 2018-08-29T06:12:15.156. More discussion here: [stackoverflow: how to get iso 8601](https://stackoverflow.com/questions/3914404/how-to-get-current-moment-in-iso-8601-format-with-date-hour-and-minute).
+     
+2. The system should expose REST APIs to perform the following operations:
+   * CUD operations ([what is CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete)) for News and Comment. If new tags and authors are passed during creation/modification – they should be created in the DB. For update operation – update only fields, that pass in request, others should not be updated. Batch insert is out of scope.
```
Comment (zarutski) [inline stage #4/module #2. REST API Advanced/rest_api_advanced.md:33]: typo on "CUD operations"

File: stage #4/module #2. REST API Advanced/rest_api_advanced.md [hunk 3]
```diff
@@ -18,105 +18,124 @@
...
+     All values should look like more -or-less meaningful: random words, but not random letters
+3. Demonstrate API using Postman tool. Prepare for demo Postman collection with APIs.
+4. (Optional) Build & run application using command line
 
 ## Theoretical part
 
-Mentee should be able to answer questions during demo session.
+Mentee should be able to answer theoretical and practical questions during demo session.
 
 ## Extra Materials
```
Comment (zarutski) [inline stage #4/module #2. REST API Advanced/rest_api_advanced.md:132]: Maybe it worth provide some Swagger link as soon as we recommend to use it for documentation

Comment (Evelina02) [reply stage #4/module #2. REST API Advanced/rest_api_advanced.md:133]: added a link

File: stage #4/module #2. REST API Advanced/rest_api_advanced.md (hunk 2 above)
Comment (Evelina02) [reply stage #4/module #2. REST API Advanced/rest_api_advanced.md]: fixed

**Pull request merged**